src/sentry/static/sentry/app/components/events/interfaces/spans/span_tree.tsx

```diff
@@ -0,0 +1,665 @@
+import React from 'react';
+import styled from 'react-emotion';
+import _ from 'lodash';
+
+import space from 'app/styles/space';
+import Count from 'app/components/count';
+
+import {SpanType, SpanEntry, SentryEvent} from './types';
+import {
+  isValidSpanID,
+  toPercent,
+  boundsGenerator,
+  SpanBoundsType,
+  SpanGeneratedBoundsType,
+} from './utils';
+import {DragManagerChildrenProps} from './dragManager';
+import SpanDetail from './spanDetail';
+
+type TraceType = {
+  type: 'trace';
+  span_id: string;
+  trace_id: string;
+};
+
+type LookupType = {[span_id: string]: SpanType[]};
+
+type RenderedSpanTree = {
+  spanTree: JSX.Element;
+  numOfHiddenSpansAbove: number;
+};
+
+type SpanTreeProps = {
+  traceViewRef: React.RefObject<HTMLDivElement>;
+  event: SentryEvent;
+  dragProps: DragManagerChildrenProps;
+};
+
+class SpanTree extends React.Component<SpanTreeProps> {
+  renderSpan = ({
+    treeDepth,
+    numberOfChildrenOfParent,
+    numOfHiddenSpansAbove,
+    spanID,
+    traceID,
+    lookup,
+    span,
+    generateBounds,
+    pickSpanBarColour,
+  }: {
+    treeDepth: number;
+    numberOfChildrenOfParent: Array<number>;
+    numOfHiddenSpansAbove: number;
+    spanID: string;
+    traceID: string;
+    span: Readonly<SpanType>;
+    lookup: Readonly<LookupType>;
+    generateBounds: (bounds: SpanBoundsType) => SpanGeneratedBoundsType;
+    pickSpanBarColour: () => string;
+  }): RenderedSpanTree => {
+    const spanBarColour: string = pickSpanBarColour();
+
+    const spanChildren: SpanType[] = _.get(lookup, spanID, []);
+
+    const numOfSpanChildren = spanChildren.length;
+
+    const start_timestamp: number = span.start_timestamp;
+    const end_timestamp: number = span.timestamp;
```

Comment:
Should these be `startTimestamp` and `endTimestamp`?

Reply:
they should